phoenix-core/src/main/java/org/apache/phoenix/index/GlobalIndexChecker.java

```diff
@@ -108,6 +107,8 @@ public GlobalIndexScanner(RegionCoprocessorEnvironment env, Scan scan, RegionSca
                     QueryServicesOptions.DEFAULT_GLOBAL_INDEX_REPAIR_COUNT);
             minTimestamp = scan.getTimeRange().getMin();
             maxTimestamp = scan.getTimeRange().getMax();
+
+            metricsSource = MetricsIndexerSourceFactory.getInstance().createForIndexChecker();
```

Comment:
This should be a member of GlobalIndexChecker (not GlobalIndexScanner).

Reply:
It can be. However, currently GlobalIndexChecker has no member fields and entirity of the logic is in GlobalIndexScanner. Also, since the scanner object does not refer back to the checker, the metricsSource would have to be passed via constructor and still be a member of the scanner. Let me know if you feel strongly, and I will do the change.

Reply:
It does not have a member field in the master branch but has it in the 4.x branches. You do not need a constructor for a RegionObserver but need to implement the start method where you can initialize it. You can just follow what IndexRegionObsever does. Please note that GlobalIndexChecker object is created when the table is loaded whereas GlobalIndexScanner is created for every scan. This is the reason I suggested to move the field to the GlobalIndexChecker class.

Reply:
Got it. Will do as suggested. Thanks for providing context!